engine/table/src/main/java/io/deephaven/engine/table/impl/select/analyzers/SelectAndViewAnalyzer.java

```diff
@@ -519,195 +851,169 @@ public void close() {
+        private final Runnable[] runners;
+        private final Runnable onSuccess;
+        private final Consumer<Exception> onError;
 
-    public abstract void startTrackingPrev();
+        private volatile boolean updateComplete;
 
-    /**
-     * Was the result internally flattened? Only the STATIC_SELECT case flattens the result. If the result preserves any
-     * columns, then flattening is not permitted. Because all the other layers cannot internally flatten, the default
-     * implementation returns false.
-     */
-    public boolean flattenedResult() {
-        return false;
-    }
+        public UpdateScheduler(
+                final Runnable[] runners,
+                final Runnable onSuccess,
+                final Consumer<Exception> onError) {
+            this.runners = runners;
+            this.onSuccess = onSuccess;
+            this.onError = onError;
+        }
 
-    /**
-     * Have the column sources already been flattened? Only the STATIC_SELECT case flattens the result. A static flatten
-     * layer is only added if SelectColumn depends on an intermediate result.
-     */
-    public boolean alreadyFlattenedSources() {
-        return false;
-    }
+        public void onLayerComplete(final int layerIndex) {
+            synchronized (remainingLayers) {
+                remainingLayers.set(layerIndex, false);
+            }
 
-    /**
-     * Return the layerIndex for a given string column.
-     *
-     * <p>
-     * This is executed recursively, because later columns in a select statement hide earlier columns.
-     * </p>
-     *
-     * @param column the name of the column
-     *
-     * @return the layerIndex
-     */
-    abstract int getLayerIndexFor(String column);
+            tryToKickOffWork();
+        }
 
-    /**
-     * Can all of our columns permit parallel updates?
-     */
-    abstract public boolean allowCrossColumnParallelization();
+        private void tryToKickOffWork() {
+            needsRun.set(true);
+            while (true) {
+                if (runLock.isHeldByCurrentThread() || !runLock.tryLock()) {
+                    // do not permit re-entry or waiting on another thread doing exactly this work
+                    return;
+                }
 
-    /**
-     * A class that handles the completion of one select column. The handlers are chained together; all downstream
-     * dependencies may execute when a column completes.
-     */
-    public static abstract class SelectLayerCompletionHandler {
-        /**
-         * Note that the completed columns are shared among the entire chain of completion handlers.
-         */
-        private final BitSet completedColumns;
-        private final SelectLayerCompletionHandler nextHandler;
-        private final BitSet requiredColumns;
-        private volatile boolean fired = false;
+                try {
+                    if (needsRun.compareAndSet(true, false)) {
+                        doKickOffWork();
```

Comment:
You can use a volatile boolean. No need for atomic, nothing can race to set it false.
```suggestion
                    if (needsRun) {
                        needsRun = false;
                        doKickOffWork();
                    }
```